src/caffe/proto/caffe.proto

```diff
@@ -935,6 +935,10 @@ message PoolingParameter {
   // If global_pooling then it will pool over the size of the bottom by doing
   // kernel_h = bottom->height and kernel_w = bottom->width
   optional bool global_pooling = 12 [default = false];
+  // If dynamic_ave_pool_size is set to true, every iteration of AVE pooling will only
+  // concerns valid elements: (sum of elements within kernel)/(valid elements num)
+  // Otherwise, invalid padded elements will always be counted in the denominator
+  optional bool dynamic_ave_pool_size = 13 [default = false];
```

Comment:
My last comment and the most opinion-based: is this the best name for this flag? How about `valid_ave_pooling`?  
Also, while it's good that you've explained what this does, I think that the comment should make it more clear that _valid pooling_ ignores the padding elements in the calculation.